src/azure-cli/azure/cli/command_modules/containerapp/_archive_utils.py

```diff
@@ -0,0 +1,236 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=consider-using-f-string, consider-using-with, no-member
+
+import tarfile
+import os
+import re
+import codecs
+from io import open
+import requests
+from knack.log import get_logger
+from msrestazure.azure_exceptions import CloudError
+from azure.cli.core.azclierror import (CLIInternalError)
+from azure.cli.core.profiles import ResourceType, get_sdk
+from azure.cli.command_modules.acr._constants import TASK_VALID_VSTS_URLS
+
+logger = get_logger(__name__)
+
+
+def upload_source_code(cmd, client,
```

Comment:
why do we need this code?

Reply:
for command `az containerapp up` --source queue_acr_build